core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.spec.ts

```diff
@@ -436,4 +436,39 @@ describe('Lesson Information card modal component', () => {
 
     expect(componentInstance.saveProgressMenuIsShown).toBeFalse();
   });
+
+  it('should return 0% when no checkpoints are completed', () => {
+    componentInstance.completedCheckpointsCount = 0;
+    componentInstance.checkpointCount = 5;
+    const progressPercentage = componentInstance.getProgressPercentage();
+    expect(progressPercentage).toEqual('0');
+  });
+
+  it('should return 100% when all checkpoints are completed', () => {
+    componentInstance.completedCheckpointsCount = 5;
+    componentInstance.checkpointCount = 5;
+    const progressPercentage = componentInstance.getProgressPercentage();
+    expect(progressPercentage).toEqual('100');
+  });
+
+  it('should return the correct percentage for 25% progress', () => {
+    componentInstance.completedCheckpointsCount = 1;
+    componentInstance.checkpointCount = 4;
+    const progressPercentage = componentInstance.getProgressPercentage();
+    expect(progressPercentage).toEqual('25');
+  });
+
+  it('should return the correct percentage for 75% progress', () => {
```

Comment:
Do we need to test 25% and 75%?

Reply:
@gp201, It is not necessary but it is good if at least one test is present in them. I can remove both tests if you want. let me know what should I do.

Reply:
Let's have the following tests:

- 25%
- rounds to the near number (67?)
- 100%
- 0%